src/main/java/hudson/plugins/gradle/injection/VcsRepositoryFilter.java

```diff
@@ -0,0 +1,61 @@
+package hudson.plugins.gradle.injection;
+
+import com.google.common.collect.ImmutableList;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public final class VcsRepositoryFilter {
+
+    private static final VcsRepositoryFilter EMPTY = new VcsRepositoryFilter(Collections.emptyList(), Collections.emptyList());
+
+    private static final String INCLUSION_QUALIFIER = "+:";
+    private static final String EXCLUSION_QUALIFIER = "-:";
+
+    private static final String SEPARATOR = "\n";
+
+    private final List<String> inclusion;
+    private final List<String> exclusion;
+
+    private VcsRepositoryFilter(List<String> inclusion, List<String> exclusion) {
+        this.inclusion = inclusion;
+        this.exclusion = exclusion;
+    }
+
+    public static VcsRepositoryFilter of(String rawVcsRepositoryFilter) {
+        if (rawVcsRepositoryFilter == null || rawVcsRepositoryFilter.isEmpty()) {
+            return EMPTY;
+        }
+
+        List<String> inclusionFilters = new ArrayList<>();
+        List<String> exclusionFilters = new ArrayList<>();
+
+        String[] parsedFilters = rawVcsRepositoryFilter.split(SEPARATOR);
```

Comment:
Want to trim the `rawVcsRepositoryFilter` before processing it?

Reply:
I'm not sure it will work if something like this is entered:
`+:foo    \n`

Reply:
I'm not 100% sure if it's a valid use case.

Reply:
Consider this as an input:
```
+:foo
-:bar
```

If there are additional spaces after `+:foo`, the trim of the original value won't catch them. 
Or I misunderstood your point then :)

Reply:
I meant for the whole string to trim new lines at the end first before processing it.

Reply:
Ah, that we already do